src/Components/Leaderboard/Leaderboard.tsx

```diff
@@ -0,0 +1,69 @@
+import { useEffect, useState } from "react"
+import "./Leaderboard.css"
+import axios, { AxiosError } from "axios"
+import { Id } from "../../Types/quiz"
+import { LeaderBoardResponseType, LeaderboardType } from "../../Types/leaderboard"
+import { ServerError } from "../../Types/quizContext"
+import "../../softui.css"
+
+export async function GetLeaderboard(id){
+    try{
+        const response = await axios.post<LeaderBoardResponseType>('https://cryptoquiz-backend.herokuapp.com/quiz/leaderboard',{
+            id:id
+        })
+        return response.data.leaderboard
+    }catch(error){
+        if(axios.isAxiosError(error)){
+            const serverError = (error as AxiosError<ServerError>)
+            if(serverError && serverError.response){
+                return serverError.response.data
+            }
+        }else{
+            return {errorMessage:"Something went wrong"}
+        }
+    }
+}
+
+export function Leaderboard({id}:Id){
+    const [leaderboard,setLeaderboard] = useState<LeaderboardType[]>()
+    const [error,setError] = useState<ServerError>()
+
+    useEffect(()=>{
+        (async function(){
+            const leaderboard = await GetLeaderboard(id)
```

Comment:
GetLeaderboard is a normal function so name it getLeaderboard, only keep functional components in Upper case.